### PR TITLE
SERVER-83900 Add type bracketing query benchmarks

### PR DESCRIPTION
**Pull request by henrinikku**
[sys-perf patch](https://spruce.mongodb.com/version/65833dd8d1fe077855558f06/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

### CODE REVIEW

File: src/phases/query/CollScanSimplifiablePredicate.yml [hunk 1]
```diff
@@ -421,3 +446,89 @@ Actors:
...
+    TemplateParameters:
+      Name: TypeBracketedLt
+      ActivePhase: 21
+      Filter:
+        {
+          # YAML doesn't support duplicate keys so the object generator is required here as 
+          # we want to construct a query of the form: {b: {$lt: 10, $lt: "abc", ...}}.
+          b: &TypeBracketedLtQueryFilter {^Object: {
+            withNEntries: 9,
+            havingKeys: "$lt",
```
Comment (alyacb) [inline src/phases/query/CollScanSimplifiablePredicate.yml:461]: gte, lte as well?

**Review by alyacb — APPROVED** [review]
LGTM!

File: src/phases/query/CollScanSimplifiablePredicate.yml (hunk 1 above)
Comment (henrinikku) [reply src/phases/query/CollScanSimplifiablePredicate.yml:461]: added!

**Review by alyacb — APPROVED** [review]
LGTM! Thanks, Henri!

**Comment by henrinikku** [comment]
evergreen merge

**Pull request merged**